CMakeLists.txt

```diff
@@ -8,9 +8,10 @@ target_include_directories(COBS INTERFACE include)
 zephyr_library()
 zephyr_library_sources(
     cobs.c
-    stream.c
 )
 
+zephyr_library_sources_ifdef(CONFIG_COBS_STREAM stream.c)
```

Comment:
Wouldn't it be better to just fix the warnings? I guess you're compiling with assertions disabled and your old zephyr version might not evaluate the macro arguments when that's the case.

Reply:
Since many fw have assertions disabled(Conhub, PAM) I think it would be better to add a ifdef CONFIG_ASSERT and use the `num_pulled` variable somehow if assertions are disabled